apis/v1alpha1/httproute_types.go

```diff
@@ -560,6 +574,65 @@ type HTTPRequestMirrorFilter struct {
 	Port *PortNumber `json:"port,omitempty"`
 }
 
+// HTTPRequestAuthorization defines configuration for the RequestAuthorization filter.
+type HTTPRequestAuthorization struct {
+	// URL specifies the address of the service used to perform the authorization check.
+	//
+	// Support: Extended
+	//
+	URL string `json:"url"`
+
+	// Protocol is used to define the Protocol that the gateway will use to communicate with the authorization
+	// service. If not set, it will default to "http" or inferred from the scheme in URL.
+	// Examples: "grpc","http2"
+	//
+	// Support: Extended
+	//
+	// +optional
+	// +kubebuilder:default=http
+	Protocol string `json:"protocol,omitempty"`
```

Comment:
For protocols over TLS, is there a way to tweak the server certificate validation?

Reply:
Yes, that would great to have, but I don't see any example of this in the API. I looked at the `HTTPRouteForwardTo` struct, and it doesn't seem to support any TLS settings. All the other TLS settings seem to be focused on the "listener side," not starting a connection.

Any suggestion on how can we add this to the filter?